eskip/doc.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2015 Zalando SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package eskip implements in-memory representation of skipper route
+definitions and a DSL for describing skipper route expressions, route
+definitions and complete routing tables.
+
+
+Grammar Summary
+
+A routing table is built up from 0 or more route definitions. The
+definitions are separated by ';'. A route definition contains one route
+expression prefixed by its id and a ':'.
+
+A routing table example:
+
+    catalog: Path("/*category") -> "https://catalog.example.org";
+    productPage: Path("/products/:id") -> "https://products.example.org";
+    userAccount: Path("/user/:id/*userpage") -> "https://users.example.org";
+
+    // 404
+    notfound: Any() ->
+        modPath(/.+/, "/notfound.html") -> static("/", "/var/www") ->
+        <shunt>
+
+A route expression always contains a matcher expression and a backend
+expression, and it can contain optional filter expressions. The matcher
+expression, each filter and the backend are separated by '->'. The
+filters take place between the matcher and the backend.
+
+A route expression example:
+
+    Path("/api/*resource") && Header("Accept", "application/json") ->
+        modPath("^/api", "") -> requestHeader("X-Type", "external") ->
+        "https://api.example.org"
+
+
+Matcher Expressions
+
+A matcher expression contains one or more conditions. An incoming
+request must fulfil each of them to match the route. The conditions are
+separated by '&&'.
```

Comment:
Maybe here would be a good point to add that all the regex predicates aka conditions expect re2 (with the link)